src/view_pos_transform.hpp

```diff
@@ -0,0 +1,178 @@
+    }
+
+    const_iterator begin() const noexcept
+        requires seqan3::const_iterable_range<urng_t>
+    {
+        return {*this, 0};
+    }
+
+    iterator end() noexcept
+    {
+        return {*this, size()};
+    }
+
+    const_iterator end() const noexcept
+        requires seqan3::const_iterable_range<urng_t>
+    {
+        return {*this, size()};
+    }
+
+    // Returns the number of elements in the view.
+    size_type size() noexcept
+    {
+        assert(size_transform != nullptr);
+        return (*size_transform)(urange);
+    }
+
+    size_type size() const noexcept
+        requires seqan3::const_iterable_range<urng_t>
+    {
+        assert(size_transform != nullptr);
+        return (*size_transform)(urange);
+    }
+
+    // Element access
+    reference operator[](size_type const n)
+    {
+        assert(n < size());
+        assert(pos_transform != nullptr);
+        return (*pos_transform)(urange, n);
+    }
+
+    const_reference operator[](size_type const n) const
+        requires seqan3::const_iterable_range<urng_t>
+    {
+        assert(n < size());
+        assert(pos_transform != nullptr);
+        return (*pos_transform)(urange, n);
+    }
+};
+
+
+template <typename urng_t, typename pos_transform_t, typename size_transform_t>
+view_pos_transform(urng_t &&, pos_transform_t, size_transform_t) -> view_pos_transform<std::ranges::all_view<urng_t>, pos_transform_t, size_transform_t>;
+
+struct pos_transform_fn
+{
+private:
+    static constexpr auto size_default_fn = [] (auto && urange) { return std::ranges::size(urange); };
+
+public:
+    template <typename pos_transform_t, typename size_transform_t = decltype(size_default_fn)>
+    constexpr auto operator()(pos_transform_t pos_transform, size_transform_t size_transform = size_default_fn) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, std::move(pos_transform), std::move(size_transform)};
+    }
+
+
+    template <std::ranges::range urng_t,
+              typename pos_transform_t,
+              typename size_transform_t = decltype(size_default_fn)>
+    constexpr auto operator()(urng_t && urange,
+                              pos_transform_t pos_transform,
+                              size_transform_t size_transform = size_default_fn) const
+    {
+        static_assert(std::ranges::viewable_range<urng_t>,
+                      "The range parameter to views::pos_transform cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::sized_range<urng_t>,
+                      "The range parameter to views::pos_transform must model std::ranges::sized_range.");
+        static_assert(std::ranges::random_access_range<urng_t>,
+                      "The range parameter to views::pos_transform must model std::ranges::random_access_range.");
+
+        return view_pos_transform{std::forward<urng_t>(urange), std::move(pos_transform), std::move(size_transform)};
+    }
+};
+
+namespace seqan3::views
```

Comment:
```suggestion
namespace views
```

We shouldn't add things to SeqAn's namespace!